subsys/debug/coredump/coredump_backend_flash_partition.c

```diff
@@ -325,15 +325,15 @@ static void coredump_flash_backend_start(void)
 		 * The header size is rounded up so the beginning of coredump
 		 * is aligned to write size (for easier read and seek).
 		 */
-		offset = backend_ctx.flash_area->fa_off;
-		offset += ROUND_UP(sizeof(struct flash_hdr_t),
-				   FLASH_WRITE_SIZE);
+		header_size = ROUND_UP(sizeof(struct flash_hdr_t),
+				       FLASH_WRITE_SIZE);
+		offset = backend_ctx.flash_area->fa_off + header_size;
 
 		ret = stream_flash_init(&backend_ctx.stream_ctx, flash_dev,
```

Comment:
Can you count on stream_flash_init() to do the erase for the header ? What happens if header_size > flash_page_size ?

Reply:
The whole partition is erased prior to writing the dump, the header size is aligned to a multiple of the flash's write size.